### PR TITLE
Add a TTS autoplay option in Edit Section Details and enable it by default for pre-reader courses

### PR DESCRIPTION
**Pull request by indhu-gunda**
Gif of changes in Edit Section Details Form: 
Peek 2020-06-25 19-42

### CODE REVIEW

**Comment by dmcavoy** [comment]
Thanks for working hard to get those tests passing Indhu! I have got the tests running on drone again.

File: apps/i18n/common/en_us.json [hunk 1]
```diff
@@ -732,6 +733,7 @@
   "exampleErrorMessage": "The function {functionName} has one or more examples that need adjusting. Make sure they match your definition and answer the question.",
   "examplesFailedOnClose": "One or more of your examples do not match your definition. Check your examples before closing",
   "exampleSolution": "Example Solution {number}",
+  "explainAutoplay": "When Automatic Instruction Playback is on, the instructions will be automatically read when a student opens a level. This enables students still learning to read or with difficulty reading the opportunity to participate in the lessons.",
```
Comment (clareconstantine) [inline apps/i18n/common/en_us.json]: The second sentence here is a little bit off grammatically - how about either removing "the opportunity" or replacing "enables" with "gives"?

File: dashboard/app/models/script.rb [hunk 2]
```diff
@@ -209,6 +209,14 @@ def self.maker_unit_scripts
     visible_scripts.select {|s| s.family_name == 'csd6'}
   end
 
+  def self.text_to_speech_script_ids
+    Script.all.select(&:text_to_speech_enabled?).pluck(:id)
+  end
+
+  def self.pre_reader_script_ids
+    Script.all.select(&:pre_reader_tts_level?).pluck(:id)
+  end
```
Comment (clareconstantine) [inline dashboard/app/models/script.rb:220]: these seem to be why the homepage query count tests are failing, but it looks like this is a pattern we use, and the Scripts db table isn't very big, so I think this should be fine.

**Review by clareconstantine — APPROVED** [review]
This looks good to me - just one text change suggestion in a string. And nice tests!

File: dashboard/app/models/script.rb (hunk 2 above)
Comment (dmcavoy) [reply dashboard/app/models/script.rb:220]: @indhu-gunda Go ahead and update the query count on those failing tests and lets see if everything passes :)

**Comment by dmcavoy** [comment]
@indhu-gunda Looks like you are going to need to merge in staging again and fix the merge conflict because someone updated the same line you did.

**Comment by dmcavoy** [comment]
Screen Shot 2020-07-22 at 4 52 02 PM

Hey Indhu - When I test this out locally none of the courses are showing up. Can you test locally and see if you see the same thing?

**Comment by dmcavoy** [comment]
@indhu-gunda Started drone again. It still looks like you will need to merge in the latest staging because of a merge conflict.

**Comment by dmcavoy** [comment]
Yay it's passing tests! I'm going to review the change locally and show it to our PM Amanda. Once she gives the thumbs up we will be able to merge it in :)

**Comment by dmcavoy** [comment]
@indhu-gunda Don't worry about the failure in the current drone run. It has to do with some stuff we are working out on our end. I'll let you know when to pull staging and merge again.

File: dashboard/app/controllers/script_levels_controller.rb [hunk 3]
```diff
@@ -79,6 +79,7 @@ def show
     @current_user = current_user && User.includes(:teachers).where(id: current_user.id).first
     authorize! :read, ScriptLevel
     @script = ScriptLevelsController.get_script(request)
+    @autoplay_enabled = current_user&.sections_as_student&.find_by(script_id: @script.id)&.autoplay_enabled
```
Comment (davidsbailey) [inline dashboard/app/controllers/script_levels_controller.rb]: Sorry to be a pain, but the term "autoplay" is already used in our system to refer to automatically playing the video on a level. please update this term to be more specific, e.g. `tts_autoplay_enabled` / `ttsAutoplayEnabled`.

**Comment by maureensturgeon** [comment]
Feature was moved to a new branch: https://github.com/code-dot-org/code-dot-org/pull/38956

**Pull request closed without merging**